cmd/agent/app/config.go

```diff
@@ -212,14 +216,24 @@ func (b *Builder) CreateAgent(mFactory metrics.Factory, logger *zap.Logger) (*Ag
 		b.CollectorServiceName = defaultCollectorServiceName
 	}
 
+	// Combine CollectorHostPort and CollectorHostPorts to
+	// form a FixedDiscoverer if applicable.
+	var hostPorts []string
+	hostPorts = append(hostPorts, b.CollectorHostPorts...)
+	if b.CollectorHostPort != "" {
+		hostPorts = strings.Split(b.CollectorHostPort, ",")
+	}
+	if len(hostPorts) != 0 {
+		d := discovery.FixedDiscoverer(hostPorts)
+		notifier := &discovery.Dispatcher{}
+		b.WithDiscoverer(d).WithDiscoveryNotifier(notifier)
+	}
```

Comment:
Figure out the "ideal" solution can be achieved by a customized flag. Just submit a new version. Please help take a look.